internal/command/fmt.go

```diff
@@ -591,29 +597,32 @@ func (c *FmtCommand) Synopsis() string {
 	return "Reformat your configuration in the standard style"
 }
 
-func bytesDiff(b1, b2 []byte, path string) (data []byte, err error) {
-	f1, err := os.CreateTemp("", "")
+func withTempFile(b []byte, fn func(*os.File) error) error {
+	f, err := os.CreateTemp("", "")
 	if err != nil {
-		return
+		return err
 	}
-	defer os.Remove(f1.Name())
-	defer f1.Close()
-
-	f2, err := os.CreateTemp("", "")
-	if err != nil {
-		return
+	_, err = f.Write(b)
+	if err == nil {
+		err = fn(f)
 	}
-	defer os.Remove(f2.Name())
-	defer f2.Close()
+	err = errors.Join(err, f.Close())
+	err = errors.Join(err, os.Remove(f.Name()))
+	return err
+}
 
-	f1.Write(b1)
-	f2.Write(b2)
+func bytesDiff(b1, b2 []byte, path string) (data []byte, err error) {
+	err = withTempFile(b1, func(f1 *os.File) error {
+		return withTempFile(b2, func(f2 *os.File) error {
+			data, err = exec.Command("diff", "--label=old/"+path, "--label=new/"+path, "-u", f1.Name(), f2.Name()).CombinedOutput()
```

Comment:
There is no prerequisites or dependencies section in the [fmt docs](https://opentofu.org/docs/cli/commands/fmt/). Could we add one and list `diff` there?

Reply:
Can you create an issue (or just fix PR) for this?

Reply:
Done https://github.com/opentofu/opentofu/pull/2801